### PR TITLE
replace btreemap to hashmap to improve performance

### PR DESCRIPTION
**Pull request by MikitaS**

### CODE REVIEW

File: core/Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,7 @@ log = { version = "0.4.17", default-features = false }
 gear-wasm-instrument = { path = "../utils/wasm-instrument", default-features = false }
 wasmparser = { package = "wasmparser-nostd", version = "0.91.0", default-features = false }
 hex = { version = "0.4.3", default-features = false, features = ["alloc"] }
+hashbrown = { version = "0.13.1" }
```
Comment (shamilsan) [inline core/Cargo.toml:21]:
```suggestion
hashbrown = "0.13.1"
```

File: gstd/Cargo.toml [hunk 2]
```diff
@@ -10,6 +10,7 @@ galloc = { path = "../galloc" }
 gcore = { path = "../gcore", features = ["codec"] }
 gstd-codegen = { path = "codegen" }
 gear-core-errors = { path = "../core-errors" }
+hashbrown = { version = "0.13.1" }
```
Comment (shamilsan) [inline gstd/Cargo.toml:13]:
```suggestion
hashbrown = "0.13.1"
```

File: pallets/gear-program/Cargo.toml [hunk 3]
```diff
@@ -17,6 +17,7 @@ codec = { package = "parity-scale-codec", version = "3.1.2", default-features =
 scale-info = { version = "2.1.1", default-features = false, features = ["derive"] }
 primitive-types = { version = "0.12.0", default-features = false, features = ["scale-info"] }
 log = { version = "0.4.17", default-features = false }
+hashbrown = { version = "0.13.1" }
```
Comment (shamilsan) [inline pallets/gear-program/Cargo.toml:20]:
```suggestion
hashbrown = "0.13.1"
```

**Review by shamilsan — APPROVED** [review]
Have you measured the performance increase?

**Comment by MikitaS** [comment]
> Have you measured the performance increase?

benchmarks show results better about 1%. Better results will be after changind all btrees, but it's possibel only after implementing some traits for  HashBrown HashMap

File: pallets/gear-program/Cargo.toml (hunk 3 above)
Comment (MikitaS) [reply pallets/gear-program/Cargo.toml]: done

File: gstd/Cargo.toml (hunk 2 above)
Comment (MikitaS) [reply gstd/Cargo.toml]: done

File: core/Cargo.toml (hunk 1 above)
Comment (MikitaS) [reply core/Cargo.toml]: done

File: runtime/gear/src/lib.rs [hunk 4]
```diff
@@ -99,7 +99,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     impl_name: create_runtime_str!("gear"),
     apis: RUNTIME_API_VERSIONS,
     authoring_version: 1,
-    spec_version: 110,
+    spec_version: 120,
```
Comment (breathx) [inline runtime/gear/src/lib.rs:102]: Don't bump runtime spec versions nowhere

**Review by breathx — CHANGES_REQUESTED** [review]
Also reimport hashmap in gstd::prelude please

File: runtime/gear/src/lib.rs (hunk 4 above)
Comment (MikitaS) [reply runtime/gear/src/lib.rs]: done

**Comment by MikitaS** [comment]
@breathx PTAL

**Comment by MikitaS** [comment]
> Also reimport hashmap in gstd::prelude please

Done

**Review by breathx — APPROVED** [review]
👍

**Comment by breathx** [comment]
Update rust and run pre commit please
Merge afterward

**Comment by MikitaS** [comment]
> Update rust and run pre commit please Merge afterward

I have already done it, locally all checks are passed

**Comment by MikitaS** [comment]
> Update rust and run pre commit please Merge afterward

Done, format fixed

**Pull request merged**